### PR TITLE
BFD-1784: Use Parameter Store for Sensitive Values

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
[BFD-1784](https://jira.cms.gov/browse/BFD-1784)

**User Story or Bug Summary:**
Encrypt, store, and manage sensitive values in AWS SSM Parameter Store.

---

### What Does This PR Do?
This PR introduces a new workspace-enabled terraform module that interacts with `ansible-vault` to securely manage sensitive values within VCS and store those values in AWS SSM Parameter Store as secure strings.

**Notes:**
1. There are a number of values included in the encrypted stores that were somewhat _blindly_ migrated from other secure storage, largely from the stateful terraform's various environment-specific tfvars files. There are some additional unused values derived from secure `ansible` variables as well. These will be removed in time, but I've included them for completeness today.
2. As of this writing, nothing yet depends on these parameters in AWS SSM Parameter Store. However, once this is accepted, I will immediately be taking advantage of these values as part of BFD-1838. BFD-1786 will further implement SSM Parameter Store, making it the happy path for configuration and secrets management.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:
* Verify all PR security questions and checklists have been completed and addressed.
* Please look over the included `base` module README and follow along with the viewing and editing steps for clarity and correctness

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mackec1 — APPROVED** [review]
Looks good; was curious on common vs. env-specific eyaml vis-a-vis ssh keys....they're all the same for each env

**Review by keithdadkins — CHANGES_REQUESTED** [review]
A very sensible approach that sets us on the right path. Good stuff.

Note, I did not compare config values with current, but they look ok and I was able to view all the things in all the environments. 

I'm also assuming this approach plays well with warm pools since the basic mechanism for getting values from the vaults remain largely the same? Seems like it.

Only request is to get the vault password file added to .gitignore to help prevent monkeys like me from leaking secrets.

Again, good work!

**Comment by mjburling** [comment]
> was curious on common vs. env-specific eyaml vis-a-vis ssh keys....they're all the same for each env

Fine question. I was mulling this over with Keith the other day, and we just settled on the very basic requirement to control which users gets access to sudoers vs basic ssh access. To that end:
- We define the ssh users once along with their pub keys in the non-SDLC `mgmt` environment
- Per SDLC-environment, we specify which of those ssh users ultimately gets added to sudoers.

This could use further refinement, but it's simpler and at least as good as what already exists.

File: ops/terraform/services/base/README.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+Each environment is configured in a terraform workspace, named for the environment it defines.
+There are four, known _established environments_ (`local.established_envs`) that enjoy _special_ treatment.
+Their respective configurations come from appropriately named, environment-specific yaml files found in the aforementioned values directory.
+
+To summarize, terraform reads encrypted **and** plaintext yaml-encoded _key:value_ pairs and stores environment-specific AWS SSM Parameter Store _parameter-path:parameter-value_ pairs.
+
+### Formatting and Validation
+
+As of mid July 2022, technical controls for standards enforcement are still forthcoming. As a stopgap, here are some guidelines in the spirit of keeping things simple:
+- hierarchies or paths conform to a 4-tuple prefix and leaf `/bfd/${env}/${group}/nonsensitive/${leaf}` format
```
Comment (christrotteradhoc) [inline ops/terraform/services/base/README.md:27]: sensitive for now only, right?

**Review by mackec1 — APPROVED** [review]

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**